sys/kern/exec.c

```diff
@@ -405,6 +424,9 @@ static int _do_execve(exec_args_t *args) {
   kfree(M_STR, p->p_elfpath);
   p->p_elfpath = kstrndup(M_STR, prog, PATH_MAX);
 
+  kfree(M_STR, p->p_args);
+  p->p_args = prepare_pargs(args);
+
   klog("Enter userspace with: pc=%p, sp=%p", eh.e_entry, stack_top);
   return EJUSTRETURN;
 
```

Comment:
@j-piecuch I am not concerned about this change. I wanted to have it near creating arguments for process.

Reply:
Fair enough.